distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/restito/ChunkedContent.java

```diff
@@ -0,0 +1,154 @@
+     *                         the first character. Once this number is reached, the message will
+     *                         send successfully.
+     * @return composite action that holds the headers required for a plain text response as well
+     * as the response function itself.
+     */
+    public static Action chunkedContentWithHeaders(String responseMessage, Duration messageDelay,
+            int numberOfFailures) {
+        return composite(getHeaders(),
+                chunkedContent(responseMessage, messageDelay, numberOfFailures));
+    }
+
+    /**
+     * Private inner class representing the response function
+     */
+    private static class ChunkedContentFunction implements Function<Response, Response> {
+        private String responseMessage;
+
+        private long messageDelayMs;
+
+        private int numberOfFailures;
+
+        private int numberOfRetries = 0;
+
+        /**
+         * Constructor for a response that has a delay and a number of planned failure.
+         *
+         * @param responseMessage  Message to be sent in response.
+         * @param messageDelay     Time to wait between sending each character of the message.
+         * @param numberOfFailures Number of times to fail (simulate network disconnect) after sending
+         *                         the first character. Once this number is reached, the message will
+         *                         send successfully.
+         */
+        private ChunkedContentFunction(String responseMessage, Duration messageDelay,
+                int numberOfFailures) {
+            this.responseMessage = responseMessage;
+            this.messageDelayMs = messageDelay.toMillis();
+            this.numberOfFailures = numberOfFailures;
+        }
+
+        /**
+         * Implementation of the Function interface's apply method. This class can also be used as a
+         * Function<Response, Response> directly in the Restito response if custom actions are needed.
+         *
+         * @param response
+         * @return
+         */
+        @Override
+        public Response apply(Response response) {
+            return respond(response);
+        }
+
+        private Response respond(Response response) {
+            for (char c : responseMessage.toCharArray()) {
+                try {
+                    response.getNIOWriter()
+                            .write(c);
+                    response.flush();
+                    sleep(messageDelayMs);
+
+                    // fail download by ungracefully closing the output buffer to simulate connection lost
+                    if (numberOfRetries < numberOfFailures) {
+                        response.getOutputBuffer()
+                                .recycle();
+                        numberOfRetries++;
+                        break;
+                    }
+                } catch (IOException | InterruptedException e) {
```

Comment:
1) It would probably be better to actually deal with it, if anything. (eg, retry the write, etc)
2) This is for testing, and probably has no benefit being reinterrupted in this case since we aren't constructing or adhearing to _flexible cancellation policies_ or the like.